### PR TITLE
fix(xy): consider `useDefaultGroupDomain` on scale config

### PR DESCRIPTION
**Pull request by markov00**
## Summary

The `useDefaultGroupDomain` was introduced to allow clustering of multiple series with a common Y domain.

The #1087 PR introduced a regression where the `useDefaultGroupDomain` was not considered when defined as the actual `groupId` for a specific group of series. This issue throws an error when a series use `useDefaultGroupDomain` with a groupId that doesn't exist in the chart configuration. This is an edge case but a legit one.

### Checklist

Delete any items that are not applicable to this PR.

- [x] Proper documentation or storybook story was added for features that require explanation or tutorials
- [x] Unit tests were updated or added to match the most common scenarios

### CODE REVIEW

**Review by nickofthyme — APPROVED** [review]
LGTM, tested locally.

File: stories/bar/3_with_axis.tsx [hunk 1]
```diff
@@ -37,6 +37,8 @@ export const Example = () => {
         yScaleType={ScaleType.Linear}
         xAccessor="x"
         yAccessors={['y']}
+        groupId="IDgroupId"
+        useDefaultGroupDomain="IDUseDefaultGroupDomain"
```
Comment (nickofthyme) [inline stories/bar/3_with_axis.tsx:41]: Was this intentional? If so need to update vrt to fix ci errors.

```
yarn test:integration:local --testNamePattern='With Axis' -u
```

Comment (markov00) [reply stories/bar/3_with_axis.tsx]: you are right, not intentional, I've rolled back the changes on that story: https://github.com/elastic/elastic-charts/pull/1119/commits/c0f9650f29f080e3401c1987d077b276b43ae4bb

Comment (markov00) [reply stories/bar/3_with_axis.tsx]: I've also checked with a previous commit if that was the existing behaviour (removing the axis) and it looks like is correct.

**Review by nickofthyme — APPROVED** [review]
👍🏼

**Pull request merged**